source/common/stats/timespan_impl.cc

```diff
@@ -0,0 +1,49 @@
+#include "common/stats/timespan_impl.h"
+
+namespace Envoy {
+namespace Stats {
+
+HistogramCompletableTimespanImpl::HistogramCompletableTimespanImpl(Histogram& histogram,
+                                                                   TimeSource& time_source)
+    : time_source_(time_source), histogram_(histogram), start_(time_source.monotonicTime()) {
+  ensureTimeHistogram(histogram);
+}
+
+std::chrono::milliseconds HistogramCompletableTimespanImpl::elapsed() const {
+  return HistogramCompletableTimespanImpl::elapsedDuration<std::chrono::milliseconds>();
+}
+
+void HistogramCompletableTimespanImpl::complete() { histogram_.recordValue(tickCount()); }
+
+void HistogramCompletableTimespanImpl::ensureTimeHistogram(const Histogram& histogram) const {
+  switch (histogram.unit()) {
+  case Histogram::Unit::Null:
+  case Histogram::Unit::Microseconds:
+  case Histogram::Unit::Milliseconds:
+    return;
+  case Histogram::Unit::Unspecified:
+  case Histogram::Unit::Bytes:
+    ASSERT(0);
```

Comment:
The intention here is to demonstrate that this is a programming error, right? 

If so, can you make this `ASSERT(false, "some text that is useful...")` so that in dev people will see the mistake they have made? Also, technically I think this should be a RELEASE_ASSERT since we will crash later on anyway with NOT_REACHED, right?

Reply:
Right, I've added an error message which hopefully would be useful and agree it's better as a release assert because if such error slips through the tests and debug builds it's better to fail with a clear message than panic. Fixed both.